### DISCUSSION TITLE
MRTK v3.0.0 Preview 18 Release & New Mixed Reality Toolkit Organization

### DISCUSSION THREAD

**Discussion by AMollis** [microsoft/MixedRealityToolkit-Unity#11772] [Releases]
![MRTK3_Release_Image_v3_0_0_pre_18](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/36461279/0c1604be-0e35-4d3d-bb26-2d0f172e0a5b)

We are very happy to announce the release of preview 18 under the new [Mixed Reality Toolkit](https://github.com/orgs/MixedRealityToolkit) organization. This new organization brings together affiliated corporations across the mixed reality industry to help build future versions of the Mixed Reality Toolkit (MRTK). For more information about this new organization, please visit the [Microsoft Tech Community Mixed Reality blog post](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Ftechcommunity.microsoft.com%2Ft5%2Fmixed-reality-blog%2Fmixed-reality-toolkit-3-moves-to-an-independent-organization-to%2Fba-p%2F3898941&data=05%7C01%7Camollis%40zune.net%7Cf6ecceea8390473dc0e308dba00a2451%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638279737647976200%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=jA04D2AE0Zu41e%2B9ERFWnuxsCAeuCTy3mH79TN2BWhM%3D&reserved=0).

With the new organization comes some breaking changes. The first being the removal of the _Microsoft_ prefix from all namespaces, which now start with _MixedReality.Toolkit_. Second, packages have been renamed from `com.microsoft.mrtk.*` to `org.mixedrealitytoolkit.*`. The old packages still exist but are deprecated. Applications should switch to the new packages as soon as possible. Next, the new version of MRTK has been relicensed under [BSD 3-Clause](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/LICENSE.md). This permissive license is somewhat like the previously used MIT License, with an additional non-endorsement clause. Developers should review the new [license](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/LICENSE.md) before incorporating these latest changes.

The final breaking change is a new repository. Going forward a new repository will be used for MRTK development.  The new repository is located at https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity. The old repository, https://github.com/microsoft/MixedRealityToolkit-Unity, is remaining under Microsoft's management. Microsoft is committed to the next version of MRTK, and recommends applications move to MRTK3.  However, Microsoft will continue to support and address critical MRTK2 issues, until MRTK2 is deprecated.   

Along with the breaking changes, this preview release contains additional fixes and minor additions. As always, we welcome your feedback on these changes and look forward to releasing MRTK3 GA this September. 

# Changelog for MRTK3 pre.18

## [pre.18] - 2023-8-21 - [`a194901`](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/commit/a1949013355f7b3561681faf2971647ccd4c6fc4)

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/). The content is mirrored at [the changelog page of the Microsoft docs website](https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/changelogs/mrtk3-pre-18-changelog), for now.

### Added

- A sample scene For Spatial Mouse; ([#11683](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11683))
- A `ComponentCache<T>` class has been added to help cache commonly queries scene components. ([11686](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11686))
- More sample scenes from MRTK2. ([11710](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11710), [11733](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11733), [11734](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11734), [11736](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11736), [11740](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11740), [11744](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11744), [11748](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11748))
- The `UnboundedTrackingMode` component was added to replace the soon-to-be deprecated `Microsoft.MixedReality.OpenXR.EyeLevelSceneOrigin` in Microsoft's Mixed Reality OpenXR Plugin. ([11750](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11750))
- Contribution guidelines have been published to the MRTK3 project. ([10](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/10), [17](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/17))
- More project maintainers from Qualcomm, Magic Leap, and Microsoft have been added. ([7](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/7), [18](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/18), [19](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/19), [21](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/21), [22](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/22))

### Changed

- MRTK3 now requires `com.unity.inputsystem` version 1.6.1 to address mouse input issues on HoloLens. ([11661](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11661))
- The `CameraSettingsManager` component is now on the MRTK XR Rig prefab. This component auto configures the main camera's settings based on the device's display type (`Transparent` or `Opaque`). ([11684](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11684), [11685](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11685))
- Refactored duplicate `LocateTargetHitPoint` methods to a common, shared location. ([11699](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11699))
- Moved some misplaced control prefabs to proper directories. ([11707](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11707))
- The `UnboundedTrackingMode` component was added to the MRTK XR Rig prefab. This change only impacts HoloLens 2 applications. ([11750](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11750))
- Changed how applications can create custom reticles in MRTK3. It should be easier now. ([11758](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11758), [9](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/9))
- All namespaces have had `Microsoft` removed, so to align with new MRTK organization. ([1](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/1))
- MRTK3 is now licensed under [BSD 3-Clause](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/blob/main/LICENSE.md). ([2](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/2), [30](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/30))

### Deprecated

- The `com.microsoft.mrtk.*` packages are now deprecated. Switch to the `org.mixedrealitytoolkit.*` packages instead. ([#31](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/31), [#32](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/32), [#34](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/34))

### Removed

- Removed the `SpatializationMixer` property from MRTK3 settings. This property wasn't used in the MRTK3 packages, so was removed for now. This property can be added later, once it has a well-defined use within the MRTK3 packages. ([11681](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11681), [11694](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11694))
- Removed `IVariableReticle` and replaced with `IReticleVisual` and `IVariableProgressReticle`. Depending on the scenario, applications should implement one or both of the new interfaces. ([11758](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11758), [9](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/9))

### Fixed

- MRTK3 now compiles in Unity 2022 without automated code changes. ([11668](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11668))
- *See-it-say-it* labels now appear when looking at a button. ([11677](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11677))
- *See-it-say-it* labels only appear if speech input is available. ([11677](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11677))
- The diction example scene works better with the speech recognition subsystem. ([11696](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11696))
- Bounds control handles now reappear after a game object is re-enabled. ([11698](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11698))
- Fixed XML summary block warnings. ([11706](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11706), [11719](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11719), [11723](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11723), [11724](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11724))
- Slider now works with values outside of the default range. ([11742](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11742))
- Cursor visuals are more stable. ([11739](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11739))
- The bounds control no longer "flattens" when applied to a 2D surface. ([12](https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/12))

**Comment by johannes-weidenfeller** [comment]
Just a note, I think the change of the package names is from `com.microsoft.mrtk.*` to `org.mixedrealitytoolkit.*`.

**Reply by AMollis** [reply]
Thanks for the correction.
